AdaptorsDataAccessPlugins/test/unit/src/au/gov/asd/tac/constellations/dataaccess/adaptors/providers/gaffer/GafferSimpleQueryNGTest.java

```diff
@@ -0,0 +1,138 @@
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
+import static uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser.DEFAULT_SERIALISER_CLASS_NAME;
+import uk.gov.gchq.gaffer.sketches.serialisation.json.SketchesJsonModules;
+
+import static org.mockito.Mockito.*;
+import org.openide.util.Exceptions;
+import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterable;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+
+/**
+ *
+ * @author GCHQDeveloper601
+ */
+public class GafferSimpleQueryNGTest {
+
+    public GafferSimpleQueryNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        JSONSerialiser.update(DEFAULT_SERIALISER_CLASS_NAME, SketchesJsonModules.class.getCanonicalName(), Boolean.TRUE);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of class ExtractFromContentPlugin.
+     *
+     * @throws java.lang.Exception
+     */
+    @Test
+    public void testAddResultsToRecordStore() {
+        GafferSimpleQuery simpleQuery = new GafferSimpleQuery();
+
+        final RecordStore recordStore = new GraphRecordStore();
+        List<Element> elements = fetchElementsFromFile("resources/exampleGafferResponseOneHop.json");
+        elements.forEach(e -> simpleQuery.addResultsToRecordStore(e, recordStore));
+        assertEquals(recordStore.size(), 3);
+    }
+    
+    @Test
+    public void testQueryForOneHop(){
+        final RecordStore recordStore = new GraphRecordStore();
+        GafferConnector connMock = mock(GafferConnector.class);
+        GafferSimpleQuery simpleQuery = new GafferSimpleQuery();
+        
+        List<String> queryIds = Arrays.asList("M4");
+        var opChain =simpleQuery.buildOneHopChain(queryIds);
+        
+        List<Element> elements = fetchElementsFromFile("resources/exampleGafferResponseOneHop.json");
+        try {
+            when(connMock.sendQueryToGaffer(opChain)).thenReturn(elements);
+        } catch (IOException | InterruptedException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        
+        simpleQuery.setGafferConnectorService(connMock);
+        simpleQuery.fetchResults(opChain, recordStore);
+        assertEquals(recordStore.size(), 3);
+    }
+    
+    @Test
+    public void testQueryForTwoHop(){
+        final RecordStore recordStore = new GraphRecordStore();
+        GafferConnector connMock = mock(GafferConnector.class);
+        GafferSimpleQuery simpleQuery = new GafferSimpleQuery();
+        
+        List<String> queryIds = Arrays.asList("M4");
+        var opChain =simpleQuery.buildTwoHopChain(queryIds);
```

Comment:
Probably less important for tests, but my previous comment about unclear types also applies here.

Reply:
Done